core/trie/node.go

```diff
@@ -138,3 +139,62 @@ func (n *Node) UnmarshalBinary(data []byte) error {
 	n.RightHash.SetBytes(data[:felt.Bytes])
 	return nil
 }
+
+func (n *Node) String() string {
+	return fmt.Sprintf("Node{Value: %s, Left: %s, Right: %s, LeftHash: %s, RightHash: %s}", n.Value, n.Left, n.Right, n.LeftHash, n.RightHash)
+}
+
+// Update the receiver with non-nil fields from the `other` Node.
+// If a field is non-nil in both Nodes, they must be equal, or an error is returned.
+//
+// This method modifies the receiver in-place and returns an error if any field conflicts are detected.
+//
+//nolint:gocyclo
+func (n *Node) Update(other *Node) error {
```

Comment:
If `other` contains a non-nil `value`, but conflicting `LeftHashes`, wouldn't the `receiver`s `value` be incorrectly updated?`

Reply:
Ahh yes thanks for spotting. Fixed this part.